### ISSUE TITLE
Batchsize and learning rate scheduler

### ISSUE THREAD

**Issue by sachinruk** [PyTorchLightning/pytorch-lightning#2903]
I was wondering if I need to adjust the batch size when using TPUs. I had a memory error when trying to run a (image 256x256x3) batch of size 128, which works perfectly fine on GPUs. 

Furthermore, do I need to adjust my custom learning rate scheduler which on GPUs run every batch (not just epoch):
```
def configure_optimizers(self):
        optimizer = get_optimizer(self)
        one_cycle_scheduler = OneCycleScheduler(optimizer, EPOCHS * steps_per_epoch // GRAD_ACCUMULATE)
        scheduler = {'scheduler': one_cycle_scheduler, "interval": "step"}

        return [optimizer], [scheduler]
```

pl version: 0.8.5
system: kaggle

**Comment by ananyahjha93** [comment]
@sachinruk Can you list the parameters being passed to the trainer in both the cases?

**Comment by sachinruk** [comment]
```
early_stopping = EarlyStopping('val_loss')
trainer = Trainer(tpu_cores=8, max_epochs=1, val_check_interval=0.5, early_stop_callback=early_stopping, accumulate_grad_batches=GRAD_ACCUMULATE)
trainer.fit(model)
```
where `GRAD_ACCUMULATE = 1`.  Only difference is for gpus, I have `gpus=1` inside the `Trainer` instead. Let me know if there is anything else needed. Oh and its a TPUv3-8 on kaggle if that makes a difference.

**Comment by ananyahjha93** [comment]
@sachinruk I couldn't find any known bugs where the batch size doesn't translate directly between a GPU and TPU because ideally, a TPUv3-8 should have more memory. Is the memory error an OOM? Can you post the error message and code to replicate this?

**Comment by rohitgr7** [comment]
I think setting batch_size=128 will actually be 128*8 here since it's running on all tpu-cores. I don't think TPUv3-8 can handle this.
https://pytorch-lightning.readthedocs.io/en/latest/trainer.html#tpu-cores

**Comment by sachinruk** [comment]
ok I've made the kernel public: https://www.kaggle.com/sachin/torch-lightning-3-tpu. Hopefully this tells you what i'm doing wrong. I reduced the batchsize to 16 but that doesn't seem to have helped. Mind you it might be another problem. It did work on GPU without issues so I'm a bit stumped on this.

**Comment by sachinruk** [comment]
Hey people, would love to hear your thoughts on this. Is the fact that I'm using a 'weird' lr_scheduler possibly the issue?

**Comment by rohitgr7** [comment]
Can't clone the notebook now since the competition is going to end soon. Although I don't think there is a problem with lr_scheduler there. If you still think it is then I suggest remove it and check.

**Comment by ananyahjha93** [comment]
@sachinruk any updates on this one?

**Comment by rohitgr7** [comment]
I think there is some issue with earlystopping when using TPUs, because the error is coming there only.

**Comment by lezwon** [comment]
> I think setting batch_size=128 will actually be 128*8 here since it's running on all tpu-cores. I don't think TPUv3-8 can handle this.
> https://pytorch-lightning.readthedocs.io/en/latest/trainer.html#tpu-cores

I don't think this is the issue. According to TPU docs:
_If you are using a training batch size of n, this is split, or sharded, across 8 TPU cores. If n = 128 and you are comparing a TPU to a single P100, then the GPU might be running at 70% of peak FLOPs while the TPU is only running at 20% of peak FLOPs (because the sharded TPU batch size is really 16, (128 / 8)) rather than 128. In this example, a TPU batch size of 1024 would make better use of TPU memory._

Ref: https://cloud.google.com/tpu/docs/performance-guide#model_processing_performance

**Comment by rohitgr7** [comment]
Yeah the issue was in earlystopping callback.
